Config options for `matchElementWidth` and `selectOn`

jessehouchins opened the pull request:
This PR add the `matchElementWidth` setting will auto adjust the width of the list when opening the dropdown to match the target element. It also exposes a `resize` API method to resize manually.

This PR also replaces the `selectOnFocus` setting with `selectOn` which allows for more flexibility when deciding when to open the list. The original API is still supported and the functional default is the same (selectOn: 'focus'). This was added specifically to allow `selectOn: 'click'` to behave more like a browser select element.

jonthornton commented on jquery.timepicker.js:
```diff
@@ -1148,7 +1168,7 @@
 		timeFormat: 'g:ia',
 		scrollDefault: null,
-		selectOnBlur: false,
+		selectOn: 'focus',
```
I don't think you meant to change this.

jonthornton commented:
Couple things that need to be fixed:

- No need for `matchElementWidth`. See #399.
- API changes must be backwards compatible. You can add a `showOn` option, but you can't remove `showOnFocus`.

jessehouchins commented:
Fixed the API issues...

A width option IS necessary though since I am using a percentage based width for the input element AND `appendTo: 'body'`. That combination means the dropdown width cannot match the input using CSS alone. I updated the resize method to use `list.outerWidth(self.outerWidth())` similar to #399, but possibly more appropriate. I didn't realize `outerWidth` was also a setter.

jonthornton commented:
I've merged in the `showOn` code. I still don't understand the width option. Can you walk me through why `appendTo: 'body'` is needed?

jessehouchins commented:
Sure. The `appendTo` option places the dropdown markup wherever you want in the DOM. Most often, if you move it, you'll place it in the body element. This is done if you have containers within your page that are scrollable so the dropdown will not get cut off. The example below would get cut of without it:

![screen shot 2015-06-15 at 8 20 27 am](https://cloud.githubusercontent.com/assets/402873/8160714/78c9eed4-1337-11e5-8b6b-fd65b69a986f.png)
 
When this technique is used, the dropdown is no longer in the same DOM context as the input. This either requires a fixed width on both the input and dropdown (the default css does this) or the width of the dropdown must be calculated by JS. As you can see from the screenshot above, I am not using the default CSS, but am setting my inputs to fit to their column widths. So I need the JS solution.

jonthornton commented:
Thanks, that was helpful - I see the problem now. Seems tangentially related to #404. I'm going to think on this for a bit to see if there's a more fundamental fix that would address both. It'll probably be a week before work and other obligations let me dive in to this. You may want to work off your fork in the meantime. Apologies for the delay but it will get done!

jessehouchins commented:
No worries.

KrunchMuffin commented:
@jessehouchins OT but I really like your styles. very clean.

xl97 commented:
(Sorry new to github.. first post actually)..  where was this left at?  I too am looking for a way that automatically sizes the dropdown width to match the input field it is associated with.  

From reading.. I understand I can manually adjust the width via style:  .ui-timepicker-wrapper

But this doesnt help much..  especially if the page is loaded in a different width itself.. (hence making the associated input fields smaller in responsive scenarios)

What/where is the 'js solution' at? a different .js file?   Doesnt exist?  Where is the .js would we modify if there is no 'legit' solution to be had?

Thanks!

jonthornton commented:
Width setting added in https://github.com/jonthornton/jquery-timepicker/commit/f670370e648130bca72c3d32bff3085e0367ab6b
